File: python/examples/wallet.py
from _env import load_dotenv, require_env, sponge_base_url
from paysponge import SpongeWallet

load_dotenv()
wallet = SpongeWallet.connect(
    api_key=require_env("SPONGE_API_KEY"),
    base_url=sponge_base_url(),
)

agent = wallet.get_agent()
addresses = wallet.get_addresses()
balances = wallet.get_balances()

print("Agent")
print({"id": agent.id, "name": getattr(agent, "name", None)})

print("Addresses")
print(addresses)

print("Balances")
print(balances)

File: typescript/src/wallet.ts
import { SpongeWallet } from "@paysponge/sdk";
import { requireEnv, spongeBaseUrl } from "./env.js";

const wallet = await SpongeWallet.connect({
  apiKey: requireEnv("SPONGE_API_KEY"),
  baseUrl: spongeBaseUrl(),
});

const agent = await wallet.getAgent();
const addresses = await wallet.getAddresses();
const balances = await wallet.getBalances();

console.log("Agent");
console.log({ id: agent.id, name: agent.name });

console.log("Addresses");
console.log(addresses);

console.log("Balances");
console.log(balances);
